src/emulator/auth/state.ts

```diff
@@ -171,10 +173,58 @@ export class ProjectState {
     } else {
       deleteProviders.push(PROVIDER_PHONE);
     }
+    if (user.mfaInfo) {
```

Comment:
To keep it consistent, we also need to go through the existing mfaEnrollmentIds and remove the ones that will no longer be present after the update. A simple yet powerful implementation to cover all cases: 1. remove all from `user.mfaInfo` (and also from index) 2. for each enrollments in `fields.mfaInfo`, add to `user` and also add to index. All of these should be done in `updateMfaEnrollments` function and we don't need this if statement here.